src/logging.cpp

```diff
@@ -404,16 +428,26 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
         }
 
         s += "] ";
-        str_prefixed.insert(0, s);
+        static auto c{m_colorman({Color::Attribute::FgBlue})};
+        str_prefixed.insert(0, c(s));
     }
 
     if (m_log_sourcelocations && m_started_new_line) {
-        str_prefixed.insert(0, "[" + RemovePrefix(source_file, "./") + ":" + ToString(source_line) + "] [" + logging_function + "] ");
+        static auto loc_file_color{m_colorman({Color::Attribute::Faint})};
+        const auto source_loc_file{"[" + RemovePrefix(source_file, "./") + ":" + ToString(source_line) + "] "};
+
+        static auto loc_func_color{m_colorman({Color::Attribute::FgGreen, Color::Attribute::Faint})};
+        const auto source_loc_func{"[" + logging_function + "] "};
+
+        str_prefixed.insert(0, loc_file_color(source_loc_file) + loc_func_color(source_loc_func));
     }
 
     if (m_log_threadnames && m_started_new_line) {
         const auto& threadname = util::ThreadGetInternalName();
-        str_prefixed.insert(0, "[" + (threadname.empty() ? "unknown" : threadname) + "] ");
+        static auto c{m_colorman({Color::Attribute::FgBlack, Color::Attribute::BgBlue})};
```

Comment:
At the very least, we shouldn't do any background coloring. This can easily degrade visibility of the information depending on what theme the user has set. This change shouldn't make any information harder to see.

| master | pr |
| --------- | --- |
| <img width="402" alt="Screen Shot 2022-09-06 at 9 07 52 PM" src="https://user-images.githubusercontent.com/23396902/188767603-dda50390-960a-4989-872e-5aeda6f5e1e8.png"> | <img width="335" alt="Screen Shot 2022-09-06 at 9 16 18 PM" src="https://user-images.githubusercontent.com/23396902/188767629-032ce879-4d8f-4ce7-8a83-6e46df69ae52.png"> |

Reply:
Thanks for the feedback, I had not thought of this.
I've just pushed a change, the thread name is now red without background.

![image](https://user-images.githubusercontent.com/22493292/188819265-4f438619-7589-4950-90d8-f7ee6f5eb36e.png)

I'm open to suggestions for the colors.

Also, if there are issues with the colors, the user can disable them with the env variable shown in the PR description. Maybe adding a bitcoind parameter may be more intuitive for the user though.